### ISSUE TITLE
shuffle significantly impact the performance

### ISSUE THREAD

**Issue by JIANG-CX** [graphdeco-inria/gaussian-splatting#79]
Hi, I've noticed that when I disable shuffling, the performance degrades significantly (I tested it on the tandt/train dataset). I also came across a comment mentioning 'Multi-res consistent random shuffling.' I'm curious about the meaning of 'Multi-res consistent' and why shuffling has such a significant impact. Thank you.

`            if shuffle:
                random.shuffle(scene_info.train_cameras)  # Multi-res consistent random shuffling
                random.shuffle(scene_info.test_cameras)  # Multi-res consistent random shuffling`

**Comment by grgkopanas** [comment]
Is it numerically worse or visually? Can you send an example?

**Comment by JIANG-CX** [comment]
I trained two models using the "tandt/train" dataset: one with shuffling enabled and the other without shuffling.

For the non-shuffling part, I modified the shuffle parameter from True to False in the code at line 25 of the "scene/init.py" file found at this URL: https://github.com/graphdeco-inria/gaussian-splatting/blob/main/scene/__init__.py#L25. Additionally, I made changes to the code at line 78 of the "train.py" file found at this URL: https://github.com/graphdeco-inria/gaussian-splatting/blob/main/train.py#L78. The modified code is as follows:

`viewpoint_cam = viewpoint_stack.pop()`

I then evaluated the training images using each method and obtained the following metrics and the corresponding "00000.png":

1. Shuffle:
SSIM: 0.7274160
PSNR: 20.7253075
![00000](https://github.com/graphdeco-inria/gaussian-splatting/assets/64529718/e593e520-4960-403b-9abf-b0af2f24b663)

2. No Shuffle:
SSIM: 0.5339764
PSNR: 16.2236824
![00000](https://github.com/graphdeco-inria/gaussian-splatting/assets/64529718/8513b2b5-369b-4c3b-b907-61b9551870f6)

**Comment by grgkopanas** [comment]
I would assume that if you do not shuffle the order of the images while you are training with SGD you take images in an order that makes the correlated spatially, this is not a very good property for SGD which in theory converges when you provide I.I.D samples.

Without claiming that shuffling the images suddenly makes them iid I would claim that they have better properties that not shuffling them and giving them in the order that they were capture in every epoch.

**Comment by JIANG-CX** [comment]
Thank you for your response. May I assume that it is difficult to accurately determine the position of a point from only two views that are closely spaced?

**Comment by Snosixtyboo** [comment]
> Thank you for your response. May I assume that it is difficult to accurately determine the position of a point from only two views that are closely spaced?

Yes that makes sense, since there is a lot of ambiguity on any axis that is not aligned with the image planes.
